fletcher/string_array.py

```diff
@@ -2,24 +2,31 @@
 
 from __future__ import absolute_import, division, print_function
 
-from pandas.core.dtypes.dtypes import ExtensionDtype
+import abc
 
 import numpy as np
 import pandas as pd
 import pyarrow as pa
+import six
+from pandas.core.dtypes.dtypes import ExtensionDtype
 
-from ._numba_compat import NumbaStringArray, NumbaString
-from ._algorithms import _startswith, _endswith
+from ._algorithms import _endswith, _startswith
+from ._numba_compat import NumbaString, NumbaStringArray
 from .base import FletcherArrayBase
 
 
+@six.add_metaclass(abc.ABCMeta)
 class StringDtypeType(object):
     """
     The type of StringDtype, this metaclass determines subclass ability
     """
     pass
 
 
+StringDtypeType.register(six.text_type)
```

Comment:
1. Please add a comment about what you are doing.
2. Don't register this for binary type. Binaries should be a separate DType.